src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java

```diff
@@ -122,6 +122,19 @@ public static float parseFloat(String s) throws NumberFormatException {
         return readJavaFormatString(s, BINARY_32_IX).floatValue();
     }
 
+    /**
+     * Converts a sequence of digits ('0'-'9') as well as an exponent to a positive
+     * double value
+     *
+     * @param decExp The decimal exponent of the value to generate
+     * @param digits The digits of the significand.
+     * @param length Number of digits to use
+     * @return The double-precision value of the conversion
+     */
+    public static double parseDoubleDigits(int decExp, char[] digits, int length) throws NumberFormatException {
```

Comment:
I don't think this method needs to declare `NFE` in the signature since it is not reading in any strings so it should not throw.